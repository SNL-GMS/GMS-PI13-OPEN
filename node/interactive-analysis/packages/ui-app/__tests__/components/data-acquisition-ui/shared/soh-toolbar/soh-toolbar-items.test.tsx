import {
  makeLagSortingDropdown,
  makeMissingSortingDropdown,
  makeStationLabel,
  SOHLagOptions,
  SOHMissingOptions
} from '../../../../../src/ts/components/data-acquisition-ui/shared/toolbars/soh-toolbar-items';
// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
// const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

describe('Soh toolbar items', () => {
  it('should be defined', () => {
    expect(SOHLagOptions).toBeDefined();
    expect(SOHMissingOptions).toBeDefined();
    expect(makeStationLabel).toBeDefined();
    expect(makeLagSortingDropdown).toBeDefined();
    expect(makeMissingSortingDropdown).toBeDefined();
  });

  it('function makeStationLabel should create a label', () => {
    expect(makeStationLabel('my station name')).toMatchSnapshot();
  });

  it('function makeStationLabel should create a label', () => {
    expect(makeLagSortingDropdown(SOHLagOptions.CHANNEL_FIRST, jest.fn())).toMatchSnapshot();
  });

  it('function makeStationLabel should create a label', () => {
    expect(
      makeMissingSortingDropdown(SOHMissingOptions.CHANNEL_FIRST, jest.fn())
    ).toMatchSnapshot();
  });
});
