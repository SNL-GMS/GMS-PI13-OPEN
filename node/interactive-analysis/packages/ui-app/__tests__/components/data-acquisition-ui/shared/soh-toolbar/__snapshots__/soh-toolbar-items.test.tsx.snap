// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Soh toolbar items function makeStationLabel should create a label 1`] = `
Object {
  "label": "Station Name",
  "rank": 3,
  "tooltip": "Station Name",
  "type": "LabelValue",
  "value": "my station name",
}
`;

exports[`Soh toolbar items function makeStationLabel should create a label 2`] = `
Object {
  "dropdownOptions": Object {
    "CHANNEL_FIRST": "Channel: A - Z",
    "CHANNEL_LAST": "Channel: Z - A",
    "LAG_HIGHEST": "Lag: highest to lowest",
    "LAG_LOWEST": "Lag: lowest to highest",
  },
  "label": "Sort Channels",
  "onChange": [MockFunction],
  "rank": undefined,
  "tooltip": "Sort",
  "type": "Dropdown",
  "value": "Channel: A - Z",
  "widthPx": 220,
}
`;

exports[`Soh toolbar items function makeStationLabel should create a label 3`] = `
Object {
  "dropdownOptions": Object {
    "CHANNEL_FIRST": "Channel: A - Z",
    "CHANNEL_LAST": "Channel: Z - A",
    "MISSING_HIGHEST": "Missing %: highest to lowest",
    "MISSING_LOWEST": "Missing %: lowest to highest",
  },
  "label": "Sort Channels",
  "onChange": [MockFunction],
  "rank": undefined,
  "tooltip": "Sort",
  "type": "Dropdown",
  "value": "Channel: A - Z",
  "widthPx": 220,
}
`;
